downloader/src/main/kotlin/Downloader.kt

```diff
@@ -95,13 +95,7 @@ object Downloader {
          * any automatic detection took place.
          */
         val originalVcsInfo: VcsInfo? = null
-    ) {
-        init {
-            require((sourceArtifact == null) != (vcsInfo == null)) {
-                "Either sourceArtifact or vcsInfo must be set, but not both."
-            }
-        }
-    }
+    )
```

Comment:
We should still keep a check that not *both* `sourceArtifact` and `vcsInfo` are *set*, but allow *none* to be set. Also, the docs for these properties should be adjusted, because ` null` now not necessarily means that the other property is set, and now both could be `null`.

Reply:
Done.